### PR TITLE
Fully Document BloqadeWaveforms

### PR DESCRIPTION
**Pull request by johnzl-777**

### CODE REVIEW

**Comment by johnzl-777** [comment]
Need to fix slicing, seems like I've broken it trying to add a docstring ):

**Review by weinbe58 — APPROVED** [review]
LGTM

File: lib/BloqadeWaveforms/src/waveform.jl [hunk 1]
```diff
@@ -452,3 +517,60 @@ function sinusoidal(; duration, amplitude = one(duration))
         return amplitude * sin(2π * t)
     end
 end
+
+"""
+    function (..)(first, last)
+
```
Comment (Roger-luo) [inline lib/BloqadeWaveforms/src/waveform.jl:523]: this is nice! wasn't thinking to document this at all.

**Review by Roger-luo — APPROVED** [review]

**Comment by Roger-luo** [comment]
Very nice documentation btw!

File: lib/BloqadeWaveforms/src/waveform.jl (hunk 1 above)
Comment (johnzl-777) [reply lib/BloqadeWaveforms/src/waveform.jl:523]: Made it a goal on my end to prioritize documenting anything that gets exported 😄 

This was particularly tricky to document (trying to get the docstring to show) but I got some inspiration looking at the way you documented things in `YaoSubspaceArrayReg`!

**Pull request merged**